src/server/dfly_bench.cc

```diff
@@ -481,33 +550,59 @@ void WatchFiber(absl::Time start_time, atomic_bool* finish_signal, ProactorPool*
     // we sleep with resolution of 1s but print with lower frequency to be more responsive
     // when benchmark finishes.
     ThisFiber::SleepFor(1s);
-    absl::Time now = absl::Now();
-    if (now - last_print > absl::Seconds(5)) {
-      ClientStats client_stats;
-      pp->AwaitFiberOnAll([&](auto* p) {
-        unique_lock lk(mutex);
-        client_stats += client->stats;
-        lk.unlock();
-      });
-
-      uint64_t total_ms = (now - start_time) / absl::Milliseconds(1);
-      uint64_t period_ms = (now - last_print) / absl::Milliseconds(1);
-      uint64_t period_resp_cnt = client_stats.num_responses - num_last_resp_cnt;
-      double done_perc = double(client_stats.num_responses) * 100 / resp_goal;
-      double hitrate =
-          client_stats.hit_opportunities > 0
-              ? 100 * double(client_stats.hit_count) / double(client_stats.hit_opportunities)
-              : 0;
-      CONSOLE_INFO << total_ms / 1000 << "s: " << absl::StrFormat("%.1f", done_perc)
-                   << "% done, effective RPS(now/accumulated): "
-                   << period_resp_cnt * 1000 / period_ms << "/"
-                   << client_stats.num_responses * 1000 / total_ms
-                   << ", errors: " << client_stats.num_errors
-                   << ", hitrate: " << absl::StrFormat("%.1f", hitrate) << "%";
-
-      last_print = now;
-      num_last_resp_cnt = client_stats.num_responses;
-    }
+    pp->AwaitBrief([](auto, auto*) { client->AdjustCycle(); });
+
+    int64_t now = absl::GetCurrentTimeNanos();
+    if (now - last_print < 5000000000LL)  // 5s
+      continue;
+
+    ClientStats stats;
+    float done_max = 0;
```

Comment:
the only change besides the indentation is computation of min/max `done` metric per thread client.